api/index.js

```diff
@@ -1,27 +1,116 @@
 
+    var processRecord = function processRecord(record) {
+      var formatDate = function formatDate(date) {
+        return moment(date, 'YYYY-MM-DD').format('DD/MM/YYYY');
+      };
+
+      return {
+        'system-number': record.id,
+        surname: record.subjects.child.name.surname,
+        forenames: record.subjects.child.name.givenName,
+        dob: formatDate(record.subjects.child.dateOfBirth),
+        gender: record.subjects.child.gender,
+        'birth-place': record.location.name,
+        mother: {
+          name: record.subjects.parent1.name.fullName,
+          nee: 'Alice*',
+          'birth-place': 'Manchester*',
+          occupation: 'Engineer*'
+        },
+        father: {
+          name: record.subjects.parent2.name.fullName,
+          'birth-place': 'Croydon*',
+          occupation: 'Self-employed*'
+        },
+        registered: {
+          jointly: 'No*',
+          district: 'Norfolk*',
+          'sub-district': 'Norwich*',
+          'admin-area': 'Norfolk*',
+          // Note: The API incorrectly returns the DoB in record.date
+          date: formatDate('2008-08-09') + '*'
+        }
+      };
+    };
+
     return new Promise(function returnRecords(resolve, reject) {
 
-      var records = _.filter(stubs, function (data) {
-        return _.every(data, function (value, id) {
-          if (!attrs[id]) {
-            return true;
+      var ajaxGet = function ajaxGet(url, callback) {
+        request.get(url, function requestGet(err, res, body) {
+          if (err) {
+            reject(err);
+          } else if (res.statusCode !== 200) {
+            var statusToName = {
+              404: 'NotFoundError'
+            };
+
+            var statusError = new Error('Received status code "' + res.statusCode + '" from API');
+
+            if (statusToName[res.statusCode]) {
+              statusError.name = statusToName[res.statusCode];
+            }
+
+            reject(statusError);
+          } else {
+            try {
+              var data = JSON.parse(body);
+            } catch (error) {
+              reject(error);
+            }
+
+            if (!data) {
+              reject(new Error('Empty response from API'));
+            } else {
+              resolve({records: callback(data)});
+            }
           }
-          return value.toLowerCase().indexOf(attrs[id].toLowerCase()) !== -1;
         });
-      });
+      };
+
+      var endpoint = 'http://' + config.api.host + ':' + config.api.port +
+                     '/api/v0/events/birth';
+      var url;
+
+      if (attrs['system-number']) {
+        url = endpoint + '/' + attrs['system-number'];
 
-      if (records) {
-        resolve({records: records});
+        ajaxGet(url, function singleRecord(data) {
+          return [processRecord(data)];
+        });
       } else {
-        reject(new Error('No records available'));
+        var params = {};
+        var formatDate = function formatDate(date) {
```

Comment:
Remember variables are scoped to the outer function. Either wrap this block in a function or declare these vars at the top of the function